File: app/src/main/java/com/example/weather/WeatherResponse.java
package com.example.weather;

public class WeatherResponse {
    private Main main;
//    private List<Weather> weather;
    private String name;

    // Getters and setters

    public class Main {
        private double temp;
        private double humidity;

        // Getters and setters
    }

    public class Weather {
        private String main;
        private String description;
        private String icon;

        // Getters and setters
    }
}

File: app/src/main/java/com/example/weather/WeatherResponse.kt
package com.example.weather

class WeatherResponse {
    var location: Location? = null
    var current: Current? = null

    inner class Location {
        var name: String? = null
        var region: String? = null
        var country: String? = null
    }

    inner class Current {
        var temp_c: Double = 0.0
        var temp_f: Double = 0.0

        var condition: Condition? = null

        inner class Condition {
            var text: String? = null
            var icon: String? = null
        }
    }
}

